CHANGELOG.md

```diff
@@ -10,6 +10,11 @@
 
 - Added a current depth field to the JSON reader.
 - Added base64 encoding and decoding APIs that accept `az_span`, available from the `azure/core/az_base64.h` header.
+- Public preview version of a new set of APIs to simplify the experience using Azure IoT Plug and Play. To consume this new feature, the `az::iot::hub` CMake target has been updated. The APIs can be found in the header files: `az_iot_hub_client.h` and `az_iot_hub_client_properties.h`.
+- New samples showing how to consume the new IoT Plug and Play APIs:
+  - paho_iot_pnp_component_sample.c
+  - paho_iot_pnp_sample.c
+  - paho_iot_pnp_with_provisioning_sample.c
 
 ### Bug Fixes
```

Comment:
Something to consider, once we delete the feature branch (https://github.com/Azure/azure-sdk-for-c/tree/feature/iot_pnp), this changelog entry (which points to it) will start causing link verification failures:
https://github.com/Azure/azure-sdk-for-c/blob/main/CHANGELOG.md#breaking-changes
![image](https://user-images.githubusercontent.com/6527137/131897697-7decfc76-383b-40fb-b798-11fbfe77adec.png)


What should we do here? I wonder if we should update the link to point to the release tag instead.

@weshaggard  thoughts?

@jspaith and @danewalton are there any other lingering links to the feature branch in the repo?

Reply:
RE lingering links, a quick `gci -Recurse * | Select-String "feature/iot_pnp"` at root of this directory only shows this hit.

#1894 tracks required changelog.md changes, of this is one, but shouldn't block this PR I don't think because I need to modify changelog after text comes anyway.

* I don't have strong opinions on this comment (other than needing to clean it up of course).  We could either delete the whole sentence which is fine I think and not replace it (I don't see other places in changelog.md referencing beta releases) or I could point it at "https://github.com/Azure/azure-sdk-for-c/releases/tag/1.2.0-beta.1" and word-smith that "this is where it came from versus 'will ship`".
* I need to modify this file (post this PR) as we need final release-worthy text that Wellington will give us today.  

All this clearly needs to go in today/tomorrow, pre-release.